docker/entrypoint.sh

```diff
@@ -6,113 +6,116 @@
-
-  # Enable the Prometheus metrics endpoint.
-  if [[ "${FEATURES}" == *"prometheus"* ]]; then
-    sed -i '/# endpoint_addr = "0.0.0.0:9999" # Prometheus/s/^# //' "${ZEBRA_CONF_PATH}"
-  fi
-
-  # Enable logging to a file by setting a custom log file path.
-  if [[ -n "${LOG_FILE}" ]]; then
-    mkdir -p "$(dirname "${LOG_FILE//\"/}")"
-    chown -R "${UID}:${GID}" "$(dirname "${LOG_FILE//\"/}")"
-    sed -i 's|# log_file = ".*"|log_file = "'"${LOG_FILE//\"/}"'"|' "${ZEBRA_CONF_PATH}"
-  fi
-
-  # Enable or disable colored logs.
-  if [[ -n "${LOG_COLOR}" ]]; then
-    sed -i '/# force_use_color = true/s/^# //' "${ZEBRA_CONF_PATH}"
-    sed -i '/use_color = true/s/true/'"${LOG_COLOR//\"/}"'/' "${ZEBRA_CONF_PATH}"
-  fi
-
-  # Enable or disable logging to systemd-journald.
-  if [[ -n "${USE_JOURNALD}" ]]; then
-    sed -i '/# use_journald = true/s/^# //; s/true/'"${USE_JOURNALD//\"/}"'/' "${ZEBRA_CONF_PATH}"
-  fi
+  # Base configuration
+  cat >"${ZEBRA_CONF_PATH}" <<EOF
+[network]
+network = "${NETWORK:=Mainnet}"
+listen_addr = "0.0.0.0"
+cache_dir = "${ZEBRA_CACHE_DIR}"
+
+[state]
+cache_dir = "${ZEBRA_CACHE_DIR}"
+
+$( [[ -n ${ZEBRA_RPC_PORT} ]] && cat <<-SUB_EOF
+
+[rpc]
+listen_addr = "${RPC_LISTEN_ADDR:=0.0.0.0}:${ZEBRA_RPC_PORT}"
+enable_cookie_auth = ${ENABLE_COOKIE_AUTH:=true}
+$( [[ -n ${ZEBRA_COOKIE_DIR} ]] && echo "cookie_dir = \"${ZEBRA_COOKIE_DIR}\"" )
+SUB_EOF
+)
+
+$( [[ " ${FEATURES} " =~ " prometheus " ]] && cat <<-SUB_EOF
+
+[metrics]
+endpoint_addr = "${METRICS_ENDPOINT_ADDR:=0.0.0.0}:${METRICS_ENDPOINT_PORT:=9999}"
+SUB_EOF
+)
+
+$( [[ -n ${LOG_FILE} || -n ${LOG_COLOR} || -n ${TRACING_ENDPOINT_ADDR} || -n ${USE_JOURNALD} ]] && cat <<-SUB_EOF
+
+[tracing]
+$( [[ -n ${USE_JOURNALD} ]] && echo "use_journald = ${USE_JOURNALD}" )
+$( [[ " ${FEATURES} " =~ " filter-reload " ]] && echo "endpoint_addr = \"${TRACING_ENDPOINT_ADDR:=0.0.0.0}:${TRACING_ENDPOINT_PORT:=3000}\"" )
+$( [[ -n ${LOG_FILE} ]] && echo "log_file = \"${LOG_FILE}\"" )
+$( [[ ${LOG_COLOR} == "true" ]] && echo "force_use_color = true" )
+$( [[ ${LOG_COLOR} == "false" ]] && echo "use_color = false" )
+SUB_EOF
+)
+
+$( [[ -n ${MINER_ADDRESS} ]] && cat <<-SUB_EOF
+
+[mining]
+miner_address = "${MINER_ADDRESS}"
+SUB_EOF
+)
+EOF
+}
 
-  # Set a mining address.
-  if [[ -n "${MINER_ADDRESS}" ]]; then
-    sed -i '/# miner_address = ".*"/{s/^# //; s/".*"/"'"${MINER_ADDRESS//\"/}"'"/}' "${ZEBRA_CONF_PATH}"
-  fi
+# Helper function
+exit_error() {
+  echo "$1" >&2
+  exit 1
+}
 
-  # Trim all comments and empty lines.
-  sed -i '/^#/d; /^$/d' "${ZEBRA_CONF_PATH}"
+# Creates a directory if it doesn't exist and sets ownership to specified UID:GID.
+#
+# ## Parameters
+#
+# - $1: Directory path to create and own
+create_owned_directory() {
+  local dir="$1"
+  [[ -z ${dir} ]] && return
 
-  echo "${ZEBRA_CONF_PATH}"
+  mkdir -p "${dir}" || exit_error "Failed to create directory: ${dir}"
+  chown -R "${UID}:${GID}" "${dir}" || exit_error "Failed to secure directory: ${dir}"
```

Comment:
###### non-blocking

This will create parent dirs such as `.cache` under root:

```bash
root@1cbf7683b54d:~# pwd
/home/zebra
root@1cbf7683b54d:~# ls -lah
total 12K
drwx------ 1 zebra zebra   26 Apr 14 14:10 .
drwxr-xr-x 1 root  root    10 Apr  9 13:03 ..
-rw-r--r-- 1 zebra zebra  220 Apr  9 13:03 .bash_logout
-rw-r--r-- 1 zebra zebra 3.5K Apr  9 13:03 .bashrc
drwxr-xr-x 1 root  root    16 Apr 14 14:10 .cache
drwxr-xr-x 1 root  root    34 Apr 14 14:10 .config
-rw-r--r-- 1 zebra zebra  807 Apr  9 13:03 .profile
```

Reply:
Fixed here: [fix(entrypoint): set permissions correctly for parent directories](https://github.com/ZcashFoundation/zebra/pull/9344/commits/e45c951b15993c83b1c73654779c83aa1292fa31)